src/test/java/it/pagopa/transactions/projections/handlers/v2/AuthorizationUpdateProjectionHandlerTest.java

```diff
@@ -304,4 +240,92 @@ void shouldSaveErrorCodeForNpgKOAuthorizationRequest() {
                 )
         );
     }
+
+    @Test
+    void shouldNotSaveErrorCodeForNpgKOAuthorizationRequestWhenUpdateDisabled() {
+        AuthorizationUpdateProjectionHandler handler = new AuthorizationUpdateProjectionHandler(
+                viewRepository,
+                paymentTokenValidity,
+                false
+        );
+
+        TransactionActivated transaction = TransactionTestUtils.transactionActivated(ZonedDateTime.now().toString());
+
+        Transaction expectedDocument = getRrn(transaction);
+
+        expectedDocument.setPaymentGateway(null);
+        expectedDocument.setAuthorizationCode("authorizationCode");
+        expectedDocument.setAuthorizationErrorCode("errorCode");
+        expectedDocument.setGatewayAuthorizationStatus("DECLINED");
+
+        TransactionAuthorizationCompletedData statusAuthCompleted = new TransactionAuthorizationCompletedData(
+                "authorizationCode",
+                "rrn",
+                expectedOperationTimeStamp,
+                new NpgTransactionGatewayAuthorizationData(
+                        OperationResultDto.DECLINED,
+                        "operationId",
+                        "paymentEndToEndId",
+                        "errorCode",
+                        null
+                )
+        );
+
+        TransactionAuthorizationCompletedEvent event = new TransactionAuthorizationCompletedEvent(
+                transaction.getTransactionId().value(),
+                statusAuthCompleted
+        );
+
+        TransactionActivated expected = getExpected(transaction, expectedDocument);
+
+        Mockito.when(viewRepository.findById(transaction.getTransactionId().value()))
+                .thenReturn(Mono.just(Transaction.from(transaction)));
+
+        StepVerifier.create(handler.handle(event))
+                .expectNext(expected)
+                .verifyComplete();
+
+        Mockito.verify(viewRepository, Mockito.never()).save(Mockito.any());
+    }
+
+    private TransactionActivated getExpected(
+                                             TransactionActivated transaction,
+                                             Transaction expectedDocument
+    ) {
+        return new TransactionActivated(
+                transaction.getTransactionId(),
+                transaction.getPaymentNotices(),
+                transaction.getEmail(),
+                null,
+                null,
+                ZonedDateTime.parse(expectedDocument.getCreationDate()),
+                Transaction.ClientId.CHECKOUT,
+                transaction.getTransactionActivatedData().getIdCart(),
+                paymentTokenValidity,
+                new EmptyTransactionGatewayActivationData(),
+                TransactionTestUtils.USER_ID
+        );
+
+    }
+
+    private static Transaction getRrn(TransactionActivated transaction) {
```

Comment:
what is this getRrn method means? rrn stands for "retrieval reference number (or something like that)", a payment terminology to indicate a payment identifier. here this method returns a view object, leading to misunderstandings.
Please remove this method and use just one for transaction view object creation (some from pagopa-ecommerce-commons TransactionUtils class)